core/store/tx_manager.go

```diff
@@ -670,6 +671,12 @@ func (txm *EthTxManager) handleSafe(
 	return nil
 }
 
+func updatePrometheusEthBalance(balance *assets.Eth, from common.Address) {
+	balanceFloat, err := balance.Float64()
+	logger.ErrorIf(err)
+	promETHBalance.WithLabelValues(from.Hex()).Set(balanceFloat)
+}
```

Comment:
Maybe Prometheus should not be expected to highly accurate balances?  If we run into a situation where balanceFloat cannot be converted, the logs will be flooded with probably irrelevant errors.

Reply:
It seems to handle the realistic Ethereum values so far. Also, this only fires when a transaction is confirmed, so once it gets low enough we will run out of funds and won't be able to send transactions and it won't fire.